### DISCUSSION TITLE
Migration to vue3

### DISCUSSION THREAD

**Discussion by koooge** [vuejs/vue-cli#6839] [General]
Hi there, I have some vue2 with @vue/cli projects and considering migrating to vue3 recently. It's really tough and needs to solve a complex dependency domino.
I expected [@vue/compat](https://github.com/vuejs/vue-next/tree/master/packages/vue-compat) or [vue-demi](https://github.com/vueuse/vue-demi) become a great help for my shared libraries. but it didn't work because they have similar [limitations](https://github.com/vuejs/vue-next/tree/master/packages/vue-compat#known-limitations) like VNode. According to them, vue 3.1 seems to have high compatibility with vue2. But unfortunately, @vue/cli v5 dropped support for vue 3.1 https://github.com/vuejs/vue-cli/pull/6808. 

I'm wondering if I should make my libs vue2/3 compatible or not. Or should I replace it with vite first? Do you have any good practices or recommendations?

**Comment by Viggo95** [comment]
Actually, the pr you linked is not very relevant to what you are describing

I'm already using Vue CLI 5 to build Vue3 and all you need to do is 

1. use `vue-demi` in your  shared libraries
2. Remove `vue@2` and `vue-template-compiler`
3. Install `vue@3` and `@vue/compiler-sfc` (eg. @vue/compiler-sfc@3.2.26) 
4. Modify the code , write in a way that Vue3 support

This is a tedious and time-consuming task, but  it can be finished
